qiskit/transpiler/passes/synthesis/hls_plugins.py

```diff
@@ -1047,8 +1079,88 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
 
 
 class PauliEvolutionSynthesisDefault(HighLevelSynthesisPlugin):
-    """The default implementation calling the attached synthesis algorithm."""
+    """Synthesize a PauliEvolutionGate using the default synthesis algorithm.
+
+    This plugin name is :``PauliEvolution.default`` which can be used as the key on
+    an :class:`~.HLSConfig` object to use this method with :class:`~.HighLevelSynthesis`.
+
+    The default synthesis simply calls the synthesis algorithm attached to a
+    PauliEvolutionGate.
+    """
 
     def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **options):
+        if not isinstance(high_level_object, PauliEvolutionGate):
+            # Don't do anything if a gate is called "evolution" but is not an
+            # actual PauliEvolutionGate
+            return None
+
         algo = high_level_object.synthesis
         return algo.synthesize(high_level_object)
+
+
+class PauliEvolutionSynthesisRustiq(HighLevelSynthesisPlugin):
+    """Synthesize a PauliEvolutionGate using Rustiq.
+
+    This plugin name is :``PauliEvolution.rustiq`` which can be used as the key on
+    an :class:`~.HLSConfig` object to use this method with :class:`~.HighLevelSynthesis`.
+
+    The Rustiq synthesis algorithm is described in [1], and is implemented in
+    Rust-based quantum circuit synthesis library available at
+    https://github.com/smartiel/rustiq-core.
+
+    On large circuits the plugin may take a significant runtime.
+
+    The plugin supports the following additional options:
+
+    * optimize_count (bool): if `True` the synthesis algorithm will try to optimize
+        the 2-qubit gate count; and if `False` then the 2-qubit depth.
+    * preserve_order (bool): whether the order of paulis should be preserved, up to
+        commutativity.
+    * upto_clifford (bool): if `True`, the final Clifford operator is not synthesized.
+    * upto_phase (bool): if `True`, the global phase of the returned circuit may
+        differ from the global phase of the given pauli network.
```

Comment:
so according to the docs, one can have both "upto_clifford=True" and "upto_phase=True" ?

Reply:
Yes, so if both are `False`, we want the exact synthesis -- this is the default behavior. If `upto_clifford=False` but `upto_phase` is `True`, then we can synthesize ignoring the global phase, in which case by default we use Qiskit's Clifford resynthesis, since it's usually faster and gives a very comparable quality of results. If `upto_clifford` it `True` then (as noted somewhere else) we don't care about the global phase either.